pkg/virtctl/create/vm/vm.go

```diff
@@ -1206,14 +1215,23 @@ func withVolumeSourceSnapshot(paramStr string, vm *v1.VirtualMachine) (*cdiv1.Da
 		return nil, err
 	}
 
-	if sourceStruct.Name == "" || sourceStruct.Namespace == "" {
-		return nil, params.FlagErr(VolumeImportFlag, "name and namespace are both required with Snapshot volume source")
+	if sourceStruct.Source == "" {
+		return nil, params.FlagErr(VolumeImportFlag, "src must be specified")
+	}
+
+	namespace, name, err := params.SplitPrefixedName(sourceStruct.Source)
```

Comment:
What if the user only specifies the pvc claim and not the namespace, doesn't the claim become the `namespace` and `name` is empty?

Reply:
Right, good catch. The removed check in L1209 needs to be moved to L1227.

Reply:
Actually, `params.SplitPrefixedName` has named return values and will return empty namespace + name for e.g. `stringwithoutslash`. It also ensures `name` is not empty and the the following check will ensure `namespace` is not empty. So this should be enough.

Reply:
It works exactly as Felix said, the name is checked in `SplitPrefixedName`-  https://github.com/kubevirt/kubevirt/blob/d7371feb56fcb138806b1b6e5b6ab154e900a2e0/pkg/virtctl/create/params/params.go#L170-L185 There is also one test case for this situation here - https://github.com/kubevirt/kubevirt/pull/10627/files#diff-57e5aa5a774e43ac18df44560d637ce9003cfe9f8d92fd9eaaec4d4302416832R928